cmd/api/discovery/apiresources.go

```diff
@@ -5,22 +5,40 @@ import (
 	"net/http"
 
 	"github.com/gin-gonic/gin"
-	"k8s.io/client-go/discovery"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
 
 	"github.com/kubearchive/kubearchive/cmd/api/abort"
 )
 
 // GetAPIResource set apiResource attribute in the context based on the path parameters group version and resourceType
-func GetAPIResource(discoveryInterface discovery.DiscoveryInterface) gin.HandlerFunc {
+func GetAPIResource(client rest.Interface) gin.HandlerFunc {
```

Comment:
How about adding a comment referencing the issue you opened to get DiscoveryInterface to admit a context?

Reply:
Added a paragraph for context.